gulpfile.js

```diff
@@ -1,3 +1,197 @@
+  label = label.toUpperCase();
+  message = message.toUpperCase();
+
+  let labelWidth = preferredWidthOf(label, { font: "10px Verdana" }) || 0;
+  let messageWidth =
+    preferredWidthOf(message, { font: "bold 10px Verdana" }) || 0;
+  const height = 28;
+  const hasLabel = label.length || labelColor;
+  if (labelColor == null) {
+    labelColor = "#555";
+  }
+  const horizPadding = 9;
+  const { hasLogo, totalLogoWidth, renderedLogo } = renderLogo({
+    logo,
+    badgeHeight: height,
+    horizPadding,
+    logoWidth,
+    logoPadding,
+  });
+
+  labelWidth += 10 + totalLogoWidth;
+  if (label.length) {
+    // Add 10 px of padding, plus approximately 1 px of letter spacing per
+    // character.
+    labelWidth += 10 + 2 * label.length;
+  } else if (hasLogo) {
+    if (hasLabel) {
+      labelWidth += 7;
+    } else {
+      labelWidth -= 7;
+    }
+  } else {
+    labelWidth -= 11;
+  }
+
+  // Add 20 px of padding, plus approximately 1.5 px of letter spacing per
+  // character.
+  messageWidth += 20 + 1.5 * message.length;
+  const leftWidth = hasLogo && !hasLabel ? 0 : labelWidth;
+  const rightWidth =
+    hasLogo && !hasLabel ? messageWidth + labelWidth : messageWidth;
+
+  labelColor = hasLabel || hasLogo ? labelColor : color;
+
+  color = escapeXml(color);
+  labelColor = escapeXml(labelColor);
+
+  const accessibleText = createAccessibleText({ label, message });
+
+  function renderLabelText() {
+    const textColor = "#fff";
+    const labelTextX = ((labelWidth + totalLogoWidth) / 2) * 10;
+    const labelTextLength = (labelWidth - (24 + totalLogoWidth)) * 10;
+    const escapedLabel = escapeXml(label);
+
+    const text = `<text fill="${textColor}" x="${labelTextX}" y="175" transform="scale(.1)" textLength="${labelTextLength}">${escapedLabel}</text>`;
+    return text;
+  }
+
+  function renderMessageText() {
+    const textColor = "#fff";
+
+    const text = `<text fill="${textColor}" x="${
+      (labelWidth + messageWidth / 2) * 10
+    }" y="175" font-weight="bold" transform="scale(.1)" textLength="${
+      (messageWidth - 24) * 10
+    }">
+        ${escapeXml(message)}</text>`;
+
+    return text;
+  }
+
+  return renderBadge(
+    {
+      leftWidth,
+      rightWidth,
+      accessibleText,
+      height,
+    },
+    `
+      <g shape-rendering="crispEdges">
+        <rect width="${leftWidth}" height="${height}" fill="${labelColor}"/>
+        <rect x="${leftWidth}" width="${rightWidth}" height="${height}" fill="${color}"/>
+      </g>
+      <g fill="#fff" text-anchor="middle" ${fontFamily} text-rendering="geometricPrecision" font-size="100">
+        ${renderedLogo}
+        ${hasLabel ? renderLabelText() : ""}
+        ${renderMessageText()}
+      </g>`
+  );
+}
+
+function renderBadges(cb) {
+  if (!fs.existsSync(OUTPUT_DIR)) {
+    fs.mkdirSync(OUTPUT_DIR, { recursive: true });
+  }
+  redirects.forEach((redirect) => {
+    const badge = myBadge({ message: redirect.redirect.replace(/\_/g, " ") });
```

Comment:
I wonder if we need to support overrides of the badge name. For example, in this case turning the text around makes more sense:

![image](https://user-images.githubusercontent.com/1444314/108927361-feaaab80-75f4-11eb-881e-57941bec8314.png)

Reply:
Fixed this.